challenge_spr/src/challenge_spr_states/riddle_game.py

```diff
@@ -25,141 +26,162 @@ def hear(robot, time_out):
 
-                    if action['entity'] == 'people':
-                        answer = 'In the crowd are %d people' % crowd_data['crowd_size']
+                        if action['entity'] == 'elders':
+                            answer = 'In the crowd are %d elders' % crowd_data['elders']
 
-                    if action['entity'] == 'children':
-                        answer = 'In the crowd are %d children' % crowd_data['children']
+                        if action['entity'] == 'males':
+                            answer = 'In the crowd are %d males' % crowd_data['males']
 
-                    if action['entity'] == 'adults':
-                        answer = 'In the crowd are %d adults' % crowd_data['adults']
+                        if action['entity'] == 'females':
+                            answer = 'In the crowd are %d females' % crowd_data['females']
 
-                    if action['entity'] == 'elders':
-                        answer = 'In the crowd are %d elders' % crowd_data['elders']
+                        if action['entity'] == 'men':
+                            answer = 'In the crowd are %d men' % crowd_data['men']
 
-                    if action['entity'] == 'males':
-                        answer = 'In the crowd are %d males' % crowd_data['males']
+                        if action['entity'] == 'women':
+                            answer = 'In the crowd are %d women' % crowd_data['women']
 
-                    if action['entity'] == 'females':
-                        answer = 'In the crowd are %d females' % crowd_data['females']
+                        if action['entity'] == 'boys':
+                            answer = 'In the crowd are %d boys' % crowd_data['boys']
 
-                    if action['entity'] == 'men':
-                        answer = 'In the crowd are %d men' % crowd_data['men']
+                        if action['entity'] == 'girls':
+                            answer = 'In the crowd are %d girls' % crowd_data['girls']
 
-                    if action['entity'] == 'women':
-                        answer = 'In the crowd are %d women' % crowd_data['women']
+                    # Location of placements or beacons
+                    if action['action'] == 'find_placement':
+                        entity = action['entity']
+                        locations = [l for l in common_knowledge.locations if l['name'] == entity]
+                        if len(locations) == 1:
+                            loc = locations[0]['room']
+                            answer = 'The %s can be found in the %s' % (entity, loc)
+                        else:
+                            answer = 'I dont know that object'
+
+                    # Count placements or beacons in the room
+                    if action['action'] == 'count_placement':
+                        entity = action['entity']
+                        locations = [l for l in common_knowledge.locations if l['name'] == entity]
```

Comment:
In general, avoid l, and o as variables, as they look a lot like 1 and 0 in some fonts. Better yet, use more than one letter (eg "loc" here)